libmamba/tests/test_satisfiability_error.cpp

```diff
@@ -634,4 +638,100 @@ namespace mamba
                                              create_scip,
                                              create_jupyterlab,
                                              create_double_python));
+
+    template <typename PkgRange>
+    auto create_mrepo_with_repo_meta(PkgRange const& packages,
+                                     std::string channel_name,
+                                     std::string subdir)
+    {
+        auto const tmp_dir = dir_guard(fs::temp_directory_path() / "mamba/tests"
```

Comment:
`test_satisfiability`  is not meant for testing this. Would you mind movinf these new tests to a `test_solver.cpp`, possibly moving test utility code to a common header.

Reply:
Sure, that makes a lot of sense!